instrumentation-api/build.gradle.kts

```diff
@@ -14,27 +14,23 @@ sourceSets {
       // set to generate into. By default it would be the src/main directory itself.
       srcDir("$buildDir/generated/sources/jflex")
     }
-
-    val cachingShadedDeps = project(":instrumentation-api-caching")
-    output.dir(cachingShadedDeps.file("build/extracted/shadow"), "builtBy" to ":instrumentation-api-caching:extractShadowJar")
   }
 }
 
 group = "io.opentelemetry.instrumentation"
 
 dependencies {
-  compileOnly(project(":instrumentation-api-caching"))
-
   api("io.opentelemetry:opentelemetry-api")
   api("io.opentelemetry:opentelemetry-semconv")
 
   implementation("io.opentelemetry:opentelemetry-api-metrics")
   implementation("org.slf4j:slf4j-api")
+  implementation("com.blogspot.mydailyjava:weak-lock-free")
+  implementation("com.googlecode.concurrentlinkedhashmap:concurrentlinkedhashmap-lru:1.4.2")
```

Comment:
let's shade these in. less transitive dependencies will make `instrumentation-api` more palatable for libraries to embed directly

Reply:
we copied in CLHM in order to update code to use default CHM

I copied in weak-lock-free also (instead of shading), just to avoid gradle shadow problems (only 4 files)